clusterloader/cluster_loader.go

```
@@ -93,93 +120,63 @@ var _ = framework.KubeDescribe("Cluster Loader [Feature:ManualPerformance]", fun
-	return filepath.Join(os.Getenv("GOPATH"), "src/k8s.io/perf-tests/clusterloader/content/", file)
+	return append(allNS, newNS)
 }
 
 // createTemplate does regex substitution against the template file, then creates the template
-func createTemplate(baseName string, ns *v1.Namespace, configPath string, numObjects int, tuning *clusterloaderframework.TuningSetType) {
+func createTemplate(baseName string, ns *v1.Namespace, configPath string, numObjects int, tuning *clusterloaderframework.TuningSet) error {
 	// Try to read the file
 	content, err := ioutil.ReadFile(configPath)
 	if err != nil {
-		framework.Failf("Error reading file: %s", err)
+		return err
 	}
 
 	// ${IDENTIFER} is what we're replacing in the file
-	regex, err := regexp.Compile("\\${IDENTIFIER}")
-	if err != nil {
-		framework.Failf("Error compiling regex: %v", err)
-	}
+	regex := regexp.MustCompile("\\${IDENTIFIER}")
 
 	for i := 0; i < numObjects; i++ {
 		result := regex.ReplaceAll(content, []byte(strconv.Itoa(i)))
 
 		tmpfile, err := ioutil.TempFile("", "cl")
 		if err != nil {
-			framework.Failf("Error creating new tempfile: %v", err)
+			return err
 		}
 		defer os.Remove(tmpfile.Name())
 
 		if _, err := tmpfile.Write(result); err != nil {
-			framework.Failf("Error writing to tempfile: %v", err)
+			return err
 		}
 		if err := tmpfile.Close(); err != nil {
-			framework.Failf("Error closing tempfile: %v", err)
+			return err
 		}
 
 		framework.RunKubectlOrDie("create", "-f", tmpfile.Name(), getNsCmdFlag(ns))
 		framework.Logf("%d/%d : Created template %s", i+1, numObjects, baseName)
 
 		// If there is a tuning set defined for this template
 		if tuning != nil {
-			if tuning.Templates.RateLimit.Delay != 0 {
-				framework.Logf("Sleeping %d ms between template creation.", tuning.Templates.RateLimit.Delay)
-				time.Sleep(time.Duration(tuning.Templates.RateLimit.Delay) * time.Millisecond)
+			if tuning.Templates.RateLimit.Delay != "" {
```

If we have a nil string then we're returning a nil error in `sleep()` so that will fail silently, removed the extra conditional check.